tests/rptest/tests/datalake/datalake_verifier.py

```diff
@@ -193,9 +198,13 @@ def _all_offsets_translated(self):
                     )
                     return False
 
-                if self._max_queried_offsets[p.id] < p.high_watermark - 1:
```

Comment:
Not necessarily from this PR, but wondering if it makes sense to ensure that there are no offsets higher than `_max_offset_check_override` (or some other variable)? IIUC this passes if the table contains offset 1000 but the HWM is 500, which seems possible with transactions

Reply:
I think the check makes sense in general (unrelated to this work) but I ended up revamping the whole thing with a new approach that seems more cleaner.